roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3572 @@
+  RooRealVar *param = this->getParameter(name);
+  if (!param) {
+    return;
+  }
+  if (value > param->getMax())
+    param->setMax(value);
+  if (value < param->getMin())
+    param->setMin(value);
+  param->setVal(value);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set one flag to a specific value
+
+void RooLagrangianMorphFunc::setFlag(const char *name, double value) {
+  RooRealVar *param = this->getFlag(name);
+  if (!param) {
+    return;
+  }
+  param->setVal(value);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set one parameter to a specific value and range
+
+void RooLagrangianMorphFunc::setParameter(const char *name, double value,
+                                          double min, double max) {
+  RooRealVar *param = this->getParameter(name);
+  if (!param) {
+    return;
+  }
+  param->setMin(min);
+  param->setMax(max);
+  param->setVal(value);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set one parameter to a specific value and range
+void RooLagrangianMorphFunc::setParameter(const char *name, double value,
+                                          double min, double max,
+                                          double error) {
+  RooRealVar *param = this->getParameter(name);
+  if (!param) {
+    return;
+  }
+  param->setMin(min);
+  param->setMax(max);
+  param->setVal(value);
+  param->setError(error);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// return true if the parameter with the given name is set constant, false
+/// otherwise
+
+bool RooLagrangianMorphFunc::isParameterConstant(const char *name) const {
+  RooRealVar *param = this->getParameter(name);
+  if (param) {
+    return param->isConstant();
+  }
+  return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// retrieve the RooRealVar object incorporating the parameter with the given
+/// name
+RooRealVar *RooLagrangianMorphFunc::getParameter(const char *name) const {
+
+  RooRealVar *param = dynamic_cast<RooRealVar *>(this->_operators.find(name));
+  if (!param) {
+    return NULL;
+  }
+  return param;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// retrieve the RooRealVar object incorporating the flag with the given name
+
+RooRealVar *RooLagrangianMorphFunc::getFlag(const char *name) const {
+  RooRealVar *flag = dynamic_cast<RooRealVar *>(this->_flags.find(name));
+  if (!flag) {
+    return NULL;
+  }
+  return flag;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// check if a parameter of the given name is contained in the list of known
+/// parameters
+
+bool RooLagrangianMorphFunc::hasParameter(const char *name) const {
+  RooRealVar *p = this->getParameter(name);
+  if (p)
+    return true;
+  return false;
```

Comment:
```suggestion
  return this->getParameter(name);
```
Try to be less verbose if possible, this new source file already has a lot of lines anyway!